### ISSUE TITLE
`tpm2_policysecret` does not work at all for key objects or any objects other than hierarchies

### ISSUE THREAD

**Issue by nicowilliams** [tpm2-software/tpm2-tools#3193]
`tpm2_policysecret` seems to work only for hierarchies.  The tests only test that.

Also, I think `tpm2_policysecret` for policy (non-trial) sessions needs an argument by which to refer to the session that will furnish the proof of the `authValue`.

**Comment by nicowilliams** [comment]
FYI I'm trying to write scripts that do the following:

- create a key object that has an `authValue`
- create another key object whose policy allows duplication but requires that the `authValue` of the other key be provided
- duplicate the the duplicable key (which includes satisfying its policy, which requires providing the `authValue` of the other key

I.e., I'm trying to script the process of creating a key that can be duplicated to all of a set of trusted TPMs but only by an authenticated administrator, and I don't want to have to set a hierarchy's `authValue` to act as the administrator password.

I'll try `tpm2-policysigned` as a workaround.  Hopefully I can get `tpm2_sign` to use an `authValue`.

**Comment by nicowilliams** [comment]
FYI `tpm2_policysigned` worked, and since the signer key can itself be a TPM resident key w/ an `authValue` for authorization, that means I can get the functionality I wanted from `tpm2_policysecret`.  But it'd be nice if `tpm2_policysecret` worked with transient objects -- if nothing else `tpm2_policysecret` is easier to use for this particular purpose, or would be, if it worked.

**Comment by JuergenReppSIT** [comment]
@nicowilliams  I could use policy secret for a transient primary as bind key and signing did work. Example:
```
head -c 32 /dev/urandom > /tmp/digest
tpm2_createprimary -p 123 -C o -c prim.ctx
tpm2_startauthsession --policy-session -S session.ctx
tpm2_policysecret -S session.ctx -c prim.ctx 123 -L secret.policy
tpm2_create -u key.pub -r key.priv -C prim.ctx -P 123 -L secret.policy
tpm2_load -C prim.ctx -P 123 -u key.pub -r key.priv -c key.ctx
tpm2_startauthsession --policy-session -S session2.ctx --bind-context=prim.ctx --bind-auth=123
tpm2_sessionconfig --enable-decrypt --enable-encrypt session2.ctx
tpm2_policysecret -S session2.ctx -c prim.ctx 123
tpm2_sign -c key.ctx -p session:session2.ctx -g sha256  -o sig -d /tmp/digest
tpm2_flushcontext --transient-object 
```

**Comment by JuergenReppSIT** [comment]
@nicowilliams  Sorry saw that the issue is the missing possibility to use password prompting later.

**Comment by nicowilliams** [comment]
Right, what's missing is an auth argument that refers to an HMAC session that proves knowledge of the `authValue`.

**Comment by JuergenReppSIT** [comment]
@nicowilliams I have prepared a PR that the file attribute (file:-) can also be used for tpm2_startauthsession and for the session parameter. In the following example both auth values can be used for authentication without  adding passwords to the shell commands, after PR #3199 is merged ;-) If you add the passwords directly to the shell commands it already works.
```
head -c 32 /dev/urandom > /tmp/digest
tpm2_createprimary -p 123 -C o -c prim.ctx
tpm2_startauthsession --policy-session -S session.ctx
tpm2_policysecret -S session.ctx -c prim.ctx 123 -L secret.policy
tpm2_policyauthvalue -S session.ctx  -L secret.policy
tpm2_create -u key.pub -r key.priv -C prim.ctx -P 123 -L secret.policy -p 456
tpm2_load -C prim.ctx -P 123 -u key.pub -r key.priv -c key.ctx
tpm2_startauthsession --policy-session -S session2.ctx --bind-context=prim.ctx --bind-auth=file:-
tpm2_sessionconfig --enable-decrypt --enable-encrypt session2.ctx
tpm2_policysecret -S session2.ctx -c prim.ctx 123
tpm2_policyauthvalue -S session2.ctx
tpm2_sign -c key.ctx -p session:session2.ctx+file:- -g sha256  -o sig -d /tmp/digest
tpm2_flushcontext --transient-object --loaded-session
```
